test/instances/lseth/LsETHHyperdrive.t.sol

```diff
@@ -0,0 +1,901 @@
+                fees: IHyperdrive.Fees({
+                    curve: 0,
+                    flat: 0,
+                    governanceLP: 0,
+                    governanceZombie: 0
+                })
+            });
+        uint256 contribution = 5_000e18;
+        factory.deployTarget(
+            bytes32(uint256(0xdeadbeef)),
+            deployerCoordinator,
+            config,
+            new bytes(0),
+            FIXED_RATE,
+            FIXED_RATE,
+            0,
+            bytes32(uint256(0xdeadbabe))
+        );
+        factory.deployTarget(
+            bytes32(uint256(0xdeadbeef)),
+            deployerCoordinator,
+            config,
+            new bytes(0),
+            FIXED_RATE,
+            FIXED_RATE,
+            1,
+            bytes32(uint256(0xdeadbabe))
+        );
+        factory.deployTarget(
+            bytes32(uint256(0xdeadbeef)),
+            deployerCoordinator,
+            config,
+            new bytes(0),
+            FIXED_RATE,
+            FIXED_RATE,
+            2,
+            bytes32(uint256(0xdeadbabe))
+        );
+        factory.deployTarget(
+            bytes32(uint256(0xdeadbeef)),
+            deployerCoordinator,
+            config,
+            new bytes(0),
+            FIXED_RATE,
+            FIXED_RATE,
+            3,
+            bytes32(uint256(0xdeadbabe))
+        );
+        factory.deployTarget(
+            bytes32(uint256(0xdeadbeef)),
+            deployerCoordinator,
+            config,
+            new bytes(0),
+            FIXED_RATE,
+            FIXED_RATE,
+            4,
+            bytes32(uint256(0xdeadbabe))
+        );
+        RIVER.approve(deployerCoordinator, contribution);
+        hyperdrive = factory.deployAndInitialize(
+            bytes32(uint256(0xdeadbeef)),
+            deployerCoordinator,
+            config,
+            new bytes(0),
+            contribution,
+            FIXED_RATE,
+            FIXED_RATE,
+            IHyperdrive.Options({
+                asBase: false,
+                destination: alice,
+                extraData: new bytes(0)
+            }),
+            bytes32(uint256(0xdeadbabe))
+        );
+
+        // Ensure that Alice received the correct amount of LP tokens. She should
+        // receive LP shares totaling the amount of shares that she contributed
+        // minus the shares set aside for the minimum share reserves and the
+        // zero address's initial LP contribution.
+        assertApproxEqAbs(
+            hyperdrive.balanceOf(AssetId._LP_ASSET_ID, alice),
+            contribution - 2 * hyperdrive.getPoolConfig().minimumShareReserves,
+            1e5
+        );
+
+        // Start recording event logs.
+        vm.recordLogs();
+    }
+
+    /// Deploy and Initialize ///
+
+    function test__lseth__deployAndInitialize() external {
+        // Deploy and Initialize the lsETH hyperdrive instance. Excess ether is
+        // sent, and should be returned to the sender.
```

Comment:
Similar to the comment I left in the RETH PR, this comment doesn't make sense since we aren't sending ETH.